.github/workflows/all-static-checks.yaml

```diff
@@ -61,3 +61,55 @@ jobs:
         run: |
           pip install pyyaml
           python tests/sweep_framework/sweeps_workflow_verification.py
+    name: Static analysis
+  clang-tidy:
+    runs-on: ubuntu-latest
+    permissions:
+      pull-requests: write
+      # OPTIONAL: auto-closing conversations requires the `contents` permission
+      contents: write
+    steps:
+    - uses: actions/checkout@v4
+      with:
+        ref: ${{ github.event.pull_request.head.sha }}
+        fetch-depth: 0
+        submodules: "recursive"
+    - name: Fetch base branch
+      run: |
+        git remote add upstream "https://github.com/${{ github.event.pull_request.base.repo.full_name }}"
+        git fetch --no-tags upstream "${{ github.event.pull_request.base.ref }}"
+      continue-on-error: true
+    - name: Install clang-tidy
+      run: |
+        sudo apt-get update
+        sudo apt-get install -y clang-tidy
+      continue-on-error: true
+    - name: Prepare compile_commands.json
+      run: |
+        cmake -B build -DCMAKE_EXPORT_COMPILE_COMMANDS=ON
```

Comment:
see https://github.com/tenstorrent/tt-metal/blob/main/build_metal.sh#L9

Reply:
Seems like the configure step is failing right now due to some missing dependencies.